src/NeoSharp.Core/Messaging/Handlers/AddrMessageMessageHandler.cs

```diff
@@ -12,24 +13,50 @@ public class AddrMessageMessageHandler : IMessageHandler<AddrMessage>
         #region Variables
 
         private readonly IServerContext _context;
+        private readonly IPeerFactory _factory;
 
         #endregion
 
         /// <summary>
         /// Constructor
         /// </summary>
         /// <param name="context">Context</param>
-        public AddrMessageMessageHandler(IServerContext context)
+        public AddrMessageMessageHandler(IServerContext context, IPeerFactory factory)
         {
             _context = context ?? throw new ArgumentNullException(nameof(context));
+            _factory = factory ?? throw new ArgumentNullException(nameof(factory));
         }
 
         public async Task Handle(AddrMessage message, IPeer sender)
         {
-            // TODO: do logic
+            IPEndPoint[] peers = message.Payload.Address
+                .Select(p => p.EndPoint)
+                .Where(
+                    p =>
+                    //p.Port != _context.ListenEndPoint.Port || // TODO: Check this
+                    !_context.ConnectedPeers
+                    .Where(u => u is TcpPeer)
+                    .Cast<TcpPeer>()
+                    .Any(u => u.IPEndPoint.Equals(p.Address))
+                    )
+                .ToArray();
 
-            // IPEndPoint[] peers = message.Payload.Address.Select(p => p.EndPoint).Where(p => p.Port != localNode.Port || !LocalNode.LocalAddresses.Contains(p.Address)).ToArray();
-            // if (peers.Length > 0) PeersReceived?.Invoke(this, peers);
+            if (peers.Length > 0)
+            {
+                // TODO: How can we connect here?
```

Comment:
Please, instead of extending IServerContext, implement ConnectTo method on IServer and use it.

Reply:
I can't inject IServer inside one message Handler....